# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLQuery.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLQuery.h
--- a/dom/canvas/WebGLQuery.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLQuery.h	Wed Jan 08 22:19:14 2020 +0000
@@ -17,8 +17,7 @@
 class AvailabilityRunnable;
 }  // namespace webgl
 
-class WebGLQuery final : public nsWrapperCache,
-                         public WebGLRefCountedObject<WebGLQuery>,
+class WebGLQuery final : public WebGLRefCountedObject<WebGLQuery>,
                          public LinkedListElement<WebGLQuery> {
   friend class webgl::AvailabilityRunnable;
   friend class WebGLRefCountedObject<WebGLQuery>;
@@ -39,8 +38,7 @@
 
   ////
 
-  NS_INLINE_DECL_CYCLE_COLLECTING_NATIVE_REFCOUNTING(WebGLQuery)
-  NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_NATIVE_CLASS(WebGLQuery)
+  NS_INLINE_DECL_REFCOUNTING(WebGLQuery)
 
   explicit WebGLQuery(WebGLContext* webgl);
 
@@ -51,16 +49,12 @@
   void Delete();
 
  public:
-  WebGLContext* GetParentObject() const { return mContext; }
-  virtual JSObject* WrapObject(JSContext* cx,
-                               JS::Handle<JSObject*> givenProto) override;
-
   ////
 
   void BeginQuery(GLenum target, WebGLRefPtr<WebGLQuery>& slot);
   void DeleteQuery();
   void EndQuery();
-  void GetQueryParameter(GLenum pname, JS::MutableHandleValue retval) const;
+  MaybeWebGLVariant GetQueryParameter(GLenum pname) const;
   bool IsQuery() const;
   void QueryCounter(GLenum target);
 };